portfolio/src/main/java/com/google/sps/servlets/UploadFileForm.java

```diff
@@ -0,0 +1,59 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+import java.util.*;
+
+@WebServlet("/uploadMemes")
```

Comment:
Can this be combined with the ListImagesMemes? That's normally how APIs/servers are set up. There's one servlet per "object" (in this case memes) and that servlet contains all the necessary methods (GET, POST, etc) to handle creating, getting, updating, and deleting that object. So you can have one servlet named MemeServlet.java or something that handles all requests to the url "/memes" and put your doGet() and doPost() into that servlet. Does that make sense?

Reply:
Yeah I already made it thanks for the advice